src/main/java/org/jetbrains/research/anticopypaster/models/UserSettingsModel.java

```diff
@@ -58,25 +68,64 @@ public void setComplexitySensitivity(int sensitivity){
     }
 
     public void setSizeSensitivity(int sensitivity){
+        System.out.println("We are changing the size sensitivity");
         this.sizeSensitivity = sensitivity;
         this.sizeMetrics.changeSensitivity(sensitivity);
     }    
 
     /**
-    Currently stubbed out to default to medium sens
-    This should ideally read in whatever the sensitivities are from the frontend,
-    the frontend 
+    Defaulted to medium if the user has not set up flag values,reads in
+     whatever the sensitivities are from the frontend if the user has set values
     */
     private void readSensitivitiesFromFrontend(){
+        //Default values if the user has not yet specified flag values
+        int keywordsSensFromFrontend = 2;
         int sizeSensFromFrontend = 2;
         int complexitySensFromFrontend = 2;
-        int keywordsSensFromFrontend = 2;
 
-        setComplexitySensitivity(complexitySensFromFrontend);
-        setSizeSensitivity(sizeSensFromFrontend);
+        //Grabs the sensitivity of the flags from the file if the file exists
+        File file = new File(FILE_PATH);
+        if (file.exists()) {
+            try (Scanner scanner = new Scanner(file)) {
+                //throw away first line
+                scanner.nextLine();
+                String keywordsDropdownValue = scanner.nextLine();
+                String keywordsCheckboxValue = scanner.nextLine();
+                String sizeDropdownValue = scanner.nextLine();
+                String sizeCheckboxValue = scanner.nextLine();
+                String complexityDropdownValue = scanner.nextLine();
+                String complexityCheckboxValue = scanner.nextLine();
+
+                keywordsSensFromFrontend = sensitivityInteger(keywordsDropdownValue);
+                sizeSensFromFrontend = sensitivityInteger(sizeDropdownValue);
+                complexitySensFromFrontend = sensitivityInteger(complexityDropdownValue);
+            } catch (FileNotFoundException ex) {
+                // Handle file not found exception
+            }
+        }
+
         setKeywordsSensitivity(keywordsSensFromFrontend);
+        setSizeSensitivity(sizeSensFromFrontend);
+        setComplexitySensitivity(complexitySensFromFrontend);
     }
 
+    /**
+     * Translates the flag sensitivity from a string to an int
+     * @param sensitivity - String version of the flag sensitivity
+     * @return - Integer of the string value given
+     */
+    private int sensitivityInteger(String sensitivity){
```

Comment:
Minor nitpick, we should probably be doing something similar to the parseSettingString() method in the CustomModelController where we lowercase the sensitivity string that's being passed in before checling it. Definitely not anything critical assuming the string is getting passed in with the same format as the ones being checked in the if/else, but just for consistency that'd be nice.
Could also make it a switch statement to make it more clear, granted that's also not a super big deal since the functionality is the same regardless